sympy/geometry/point3d.py

```diff
@@ -508,7 +506,7 @@ def translate(self, x=0, y=0, z=0):
 
     def transform(self, matrix):
         """Return the point after applying the transformation described
-        by the 3x3 Matrix, ``matrix``.
+        by the 4x4 Matrix, ``matrix``.
```

Comment:
I just read this:

> Fixed a bug in the transformation function. It should take a 4x4 transformation matrix, and apply it to the 3d point with a homogeneous coordinate added. Previously, the function caused the z coordinate to be zeroed.

So it sounds like this was a bug and didn't work in the first place. Can you confirm?